test/sparse/sparse.jl

```diff
@@ -1828,3 +1828,8 @@ let
     @test_throws DimensionMismatch broadcast(+, A, B, speye(N))
     @test_throws DimensionMismatch broadcast!(+, X, A, B, speye(N))
 end
+
+let A = sparse(Real[1 1])
+    @test (A + A)::SparseMatrixCSC{Int,Int} == [2 2] #19595
```

Comment:
Would you say we should verify the return type to have eltype `Int`? Presently
```Julia
julia> Real[1 1] + Real[1 1]
1×2 Array{Real,2}:
 2  2
```
so maybe `Real` would be an acceptable return type, too?